tree/treeplayer/src/TTreeProcessorMT.cxx

```diff
@@ -222,67 +222,52 @@ MakeClusters(const std::vector<std::string> &treeNames, const std::vector<std::s
 
 ////////////////////////////////////////////////////////////////////////
 /// Return a vector containing the number of entries of each file of each friend TChain
-static std::vector<std::vector<Long64_t>>
-GetFriendEntries(const std::vector<std::pair<std::string, std::string>> &friendNames,
-                 const std::vector<std::vector<std::string>> &friendFileNames)
+static std::vector<std::vector<Long64_t>> GetFriendEntries(const Internal::TreeUtils::RFriendInfo &friendInfo)
```

Comment:
As a side note, for remote files in particular, this routines is very expansive (for all the files: open the file, retrieve the TTree and then delete them) [See the difference between `TChain::GetEntriesFast` and `TChain::GetEntries for a related example]

Reply:
How should we be doing this?

Reply:
> I don't remember exactly why but that was better than having a chain that does not know its total number of entries...

Yes, that is the trade off (You either know (and pay an upfront cost to gather the information) or you don't and then pay in efficiency later ... which one wins is situation dependent (latency to the file server, download speed, contention on the server side) vs (inefficiencies of not knowing the file size)